be/src/io/fs/s3_file_writer.cpp

```diff
@@ -357,6 +365,9 @@ void S3FileWriter::_put_object(S3FileBuffer& buf) {
     DCHECK(!_closed) << "closed " << _closed;
     Aws::S3::Model::PutObjectRequest request;
     request.WithBucket(_bucket).WithKey(_key);
+    const auto& _stream_ptr = buf.get_stream();
```

Comment:
```suggestion
    const auto& stream_ptr = buf.get_stream();
```
And change `std::shared_ptr<std::iostream> S3FileWriteBufferpool::get_stream()` to `const std::shared_ptr<std::iostream>& S3FileWriteBufferpool::get_stream()`